doc/advanced_configuration.rst

```diff
@@ -397,6 +397,42 @@ The default behavior is ``sphinx_gallery_thumbnail_number = 1``. See
 :ref:`sphx_glr_auto_examples_plot_choose_thumbnail.py` for an example
 of this functionality.
 
+.. _binder_links:
+
+Generate Binder links for gallery notebooks
+===========================================
+
+Sphinx-Gallery automatically generates Jupyter notebooks for any
+examples built with the gallery. `Binder <http://mybinder.org>`_ makes it
+possible to create interactive GitHub repositories connect to cloud resources.
+If you host your documentation on a GitHub repository, it is possible to
+auto-generate a Binder link for each notebook. Clicking this link will
+take users to a live version of the Jupyter notebook where they may
+run the code interactively. For more information see the `Binder documentation
+<http://docs.mybinder.org>`_.
+
+.. warning::
+
+   Binder is still beta technology, so there may be instability in the
+   experience of users who click Binder links.
+
+In order to enable Binder links with Sphinx-Gallery, you must specify
+a few pieces of information in ``conf.py``. These are given as a nested
+dictionary following the pattern below::
+
+    sphinx_gallery_conf = {
+      ...
+      'binder': {
+         'org': '<github_org>',
+         'repo': '<github_repo>',
+         'url': '<binder_url>',  # URL serving binders (e.g. mybinder.org)
+         'branch': '<repo_branch>',  # Can also be a tag or commit hash
+         'dependencies': '<path_to_dependencies>'  # Either `requirements.txt` or `environment.yml`
```

Comment:
Is not entirely evident. I would prefer 
<path_to_dependencies_file> # either `txt` file compliant with pip syntax, or `yml` file compliant with conda environments. The filename should be arbitrary, we do the renaming.

Reply:
hmm - I guess we could assume that if any file ends in `.txt` it's a `requirements.txt` and any file that ends in `.yml` or `.yaml` is `environment.yml`...but this feels hacky to me. I was actually leaning towards letting people supply a list of filenames, since binder now supports `apt.txt`, `postBuild`, etc scripts...

Reply:
If a list of filenames is what Binder supports, putting a list of filenames here would make sense